src/scalaDebugger.ts

```diff
@@ -0,0 +1,65 @@
+import * as vscode from "vscode";
+import {
+  CancellationToken,
+  DebugConfiguration,
+  Disposable,
+  ProviderResult,
+  WorkspaceFolder
+} from "vscode";
+
+export const startAdapterCommand = "debug-adapter-start";
+export const startSessionCommand = "metals-debug-session-start";
+const configurationType = "scala";
+
+export function initialize(outputChannel: vscode.OutputChannel): Disposable[] {
+  outputChannel.appendLine("Initializing Scala Debugger");
+  return [
+    vscode.debug.registerDebugConfigurationProvider(
+      configurationType,
+      new ScalaConfigProvider()
+    )
+  ];
+}
+
+export async function start(parameters: any): Promise<Boolean> {
```

Comment:
nit: no need to make the function `async` if we're using the normal Promise api in the implementation (which I personally prefer, anyway)

```suggestion
export function start(parameters: any): Promise<Boolean> {
```